routing/missioncontrol.go

```diff
@@ -299,6 +299,22 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 	return bandwidthHints, nil
 }
 
+// UpdateEdgePolicy updates the edge policy of the routing hints kept in the
+// payment session. The number of hints updated is returned.
+func (p *paymentSession) UpdateEdgePolicy(e *channeldb.ChannelEdgePolicy) {
+
+	// run over hints and update policy if edge id matches
+	for _, edges := range p.additionalEdges {
+		for _, edge := range edges {
+			if edge.ChannelID == e.ChannelID {
```

Comment:
Are you updating the channel in both directions? I think you may also need to match the pubkey in additional edges to update only a single direction.

Reply:
The channel is updated only in one direction. That is, the node generates an error on the outgoing channel policy, and you update the hints accordingly. Am I missing something here?

Reply:
I mean, suppose you have two additional edges relating to the same channel id. They will both be updated?

Reply:
Yes. I changed it such that it takes direction into account too, such that policy will be updated only if channel ID and direction both match.
I think this also fixes a general hidden bug in routing through hint hops, not related to this PR, as additionalEdges policies didn't take into account direction. That is, the Flags were not initialized in NewPaymentSession, and routing assumes the direction is initialized in all edge policies.